### ISSUE TITLE
False positive in array_init rule with prefix operators

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#1877]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

```bash
$ Pods/SwiftLint/swiftlint lint --path file.swift                                         
Loading configuration from '.swiftlint.yml'
Linting Swift files at path file.swift
Linting 'file.swift' (1/1)
file.swift:1:1: warning: Array Init Violation: Prefer using Array(seq) than seq.map { $0 } to convert a sequence into an Array. (array_init)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.23.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods
* Paste your configuration file:

```yml
opt_in_rules:
  - array_init
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

No

* Which Xcode version are you using (check `xcode-select -p`)?

Xcode 9

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
array.map { !$0 }
```

**Comment by ArchieR7** [comment]
same as here, does anyone know how to fix this issue?

**Comment by marcelofabri** [comment]
@ArchieR7 feel free to send a PR fixing this! Let me know if you have any questions ☺️

**Comment by iDevid** [comment]
it is not in the latest release?
